app/scripts/controllers/permissions/index.js

```diff
@@ -428,15 +467,41 @@ class PermissionsController {
         // attempt to install and run the plugin, storing any errors that
         // occur during the process
         result[pluginName] = {
-          ...result[pluginName],
           ...(await this.pluginsController.processRequestedPlugin(pluginName)),
         }
       }
     }))
     return result
   }
 
+  /**
+   * Get the permitted plugins for the given origin, by plugin name.
+   * Recall that plugin names !== permission names.
+   *
+   * @param {string} origin - The origin to get the permitted plugins for.
+   */
+  getPermittedPlugins (origin) {
+
+    return this.getPermissionsFor(origin).reduce(
+      (acc, p) => {
```

Comment:
I know Aaron would make a firm stance against single-letter variables. We should probably add it to the linter, because I know it's a custom that he prefers to enforce. I think there's a good reason: They're hard to quickly text-search for.

Reply:
`p => perm`